src/renderers/shared/stack/reconciler/ReactUpdateQueue.js

```diff
@@ -58,15 +59,18 @@ function getInternalInstanceReadyForUpdate(publicInstance, callerName) {
   }
 
   if (__DEV__) {
-    warning(
-      ReactCurrentOwner.current == null,
-      '%s(...): Cannot update during an existing state transition (such as ' +
-      'within `render` or another component\'s constructor). Render methods ' +
-      'should be a pure function of props and state; constructor ' +
-      'side-effects are an anti-pattern, but can be moved to ' +
-      '`componentWillMount`.',
-      callerName
-    );
+    if (warnedOnce === false) {
+      warnedOnce = true;
+      warning(
+        ReactCurrentOwner.current == null,
```

Comment:
Also worth noting, when you have a conditional inside conditional, it's preferable to come up with something else, so having an extra layer of abstraction here could be good, as we are making the code really dirty. But we will wait for someone else to decide that. @Aweary if you wanna join, there's space for you to help too :)